lib/Telemetry/src/DispatcherFactory.h

```diff
@@ -0,0 +1,114 @@
+#include "DataQueue.h"
+#include "LogCommand.h"
+#include "Command.h"
+#include "Dispatcher.h"
+#include "Particle.h"
+#include "IntervalLogger.h"
+
+#define MAX_NUM_INTERVALS 5
+
+#ifndef _DISPTACHER_FACTORY_H_
+#define _DISPATCHER_FACTORY_H_
+
+// This isn't really a factory but it has a similar purpose:
+// Add all the commands you want to this and call build().  It
+// produces a dispatcher with all logging functionality already set up
+// You can delete this safely after build() has been called
+class DispatcherFactory {
+    public:
+        /**
+         * Constructs new DispatcherFactory with number of commands and data queue
+         * 
+         * @param numCommands The number of sensor functions you will add to the factory
+         * Note: this must be accurate and has potential to cause a segfault if the value is too low
```

Comment:
Is there any way for numCommands to be created dynamically and for more memory to be allocated as needed? Or is that too complicated (it looks like it might be). Maybe a good simple solution here would be to add make the "add" method more robust so that it won't add any more commands than the maximum? It could even print a debug message with an error if we do that by accident.

Reply:
this was just my way of avoiding having to create a dynamic array class.  I could do it but I figured it would be fine to pass this responsibility onto the user, as all the sensor getter functions are added in one place.  I agree that it would be more elegant to not have to do this

Reply:
The max_num_add is fine too

Reply:
> The max_num_add is fine too

What do you mean by this?

Reply:
I was referring to your second idea, of having a maximum number of adds the user could do.  Either works fine.  I kind of liked the idea of specifying because then we don't have to set some arbitrary amount, but I do understand that it's a bit clunky

Reply:
I'm almost tempted to make a dynamic array for our project.  Do you think it would be useful elsewhere? Something like

template<class T>
class array_t